src/engine/BMDie.php

```diff
@@ -290,7 +290,13 @@ public function roll($isTriggeredByAttack = FALSE, $isSubdie = FALSE) {
 
         if (!isset($this->value) ||
             ($this->doesReroll && !$this->has_flag('JustPerformedTripAttack'))) {
-            $this->set__value(bm_rand($this->min, $this->max));
+            $hookResultArray = $this->run_hooks(__FUNCTION__, array('die' => $this));
+
+            // if all hook results are FALSE, then roll the die
+            if (empty($hookResultArray) ||
+                (0 == count(array_filter($hookResultArray, function ($value) { return $value !== FALSE; })))) {
+                $this->set__value(bm_rand($this->min, $this->max));
+            }
```

Comment:
I don't understand why this code change has the desired effect:
* the if() statement on line 291 has a condition that either you have no value or `$this->doesReroll` has to be set
* the only time your new hooks are going to return false is when `$this->doesReroll` is *not* set

So it seems like the combination of all the new code should be that when you have a `kM` die, it exits this code with `$this->value` still unset, and freaks out.